diff -ur net-snmp-5.2.orig/agent/mibgroup/mibII/interfaces.c net-snmp-5.2/agent/mibgroup/mibII/interfaces.c
--- net-snmp-5.2.orig/agent/mibgroup/mibII/interfaces.c	2004-10-06 22:10:16.000000000 +0200
+++ net-snmp-5.2/agent/mibgroup/mibII/interfaces.c	2004-12-29 21:27:25.939529536 +0100
@@ -1383,9 +1383,9 @@
     unsigned long   rec_pkt, rec_oct, rec_err, rec_drop;
     unsigned long   snd_pkt, snd_oct, snd_err, snd_drop, coll;
     const char     *scan_line_2_2 =
-        "%lu %lu %lu %lu %*lu %*lu %*lu %*lu %lu %lu %lu %lu %*lu %lu";
+        "%llu %llu %llu %llu %*llu %*llu %*llu %*llu %llu %llu %llu %llu %*llu %llu"
     const char     *scan_line_2_0 =
-        "%lu %lu %*lu %*lu %*lu %lu %lu %*lu %*lu %lu";
+        "%llu %llu %*llu %*llu %*llu %llu %llu %*llu %*llu %llu";
 #endif
     const char     *scan_line_to_use;
     struct timeval et;                              /* elapsed time */
