--- net-snmp-5.1.1/perl/SNMP/examples/bulkwalk.pl~	Sat Apr 20 07:30:18 2002
+++ net-snmp-5.1.1/perl/SNMP/examples/bulkwalk.pl	Sat Jul 26 13:40:43 2003
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 # $Id$
 #
--- net-snmp-5.1.1/perl/manager/manager.pm.orig	2002-06-28 13:37:23.000000000 +0200
+++ net-snmp-5.1.1/perl/manager/manager.pm	2004-03-27 16:07:41.028771992 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 package NetSNMP::manager;
 
