diff -ur net-snmp-5.2.orig/net-snmp-config.in net-snmp-5.2/net-snmp-config.in
--- net-snmp-5.2.orig/net-snmp-config.in	2004-10-09 01:02:26.000000000 +0200
+++ net-snmp-5.2/net-snmp-config.in	2004-12-29 20:45:58.950609320 +0100
@@ -31,7 +31,7 @@
 exec_prefix=@exec_prefix@
 includedir=@includedir@
 libdir=@libdir@
-NSC_LDFLAGS="@LDFLAGS@"
+NSC_LDFLAGS=""
 NSC_INCLUDEDIR=${includedir}
 NSC_LIBDIR=-L${libdir}
 NSC_LIBS="@LIBS@ @WRAPLIBS@"
@@ -89,10 +89,10 @@
       ;;
     #################################################### compile
     --base-cflags)
-      echo @CFLAGS@ @CPPFLAGS@ -I${NSC_INCLUDEDIR}
+      echo @CPPFLAGS@ -I${NSC_INCLUDEDIR}
       ;;
     --cflags|--cf*)
-      echo @CFLAGS@ @DEVFLAGS@ @CPPFLAGS@ -I. -I${NSC_INCLUDEDIR}
+      echo @DEVFLAGS@ @CPPFLAGS@ -I. -I${NSC_INCLUDEDIR}
       ;;
     --srcdir)
       echo $NSC_SRCDIR
@@ -445,7 +445,7 @@
         echo "Ack.  Can't create $tmpfile."
 	exit 1
       fi
-      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS -L$NSC_EXEC_PREFIX/lib -lnetsnmpagent -lnetsnmphelpers -lnetsnmpmibs -lnetsnmp $NSC_AGENTLIBS $ldflags"
+      cmd="@CC@ $cflags @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS -L$NSC_EXEC_PREFIX/lib -lnetsnmpagent -lnetsnmphelpers -lnetsnmpmibs -lnetsnmp $NSC_AGENTLIBS $ldflags"
       echo "running: $cmd"
       `$cmd`
       if test "x$norm" != "x1" ; then
