--- net-snmp-5.6/agent/Makefile.in.orig	2010-06-08 23:05:11.000000000 +0200
+++ net-snmp-5.6/agent/Makefile.in	2010-10-24 10:33:36.007741043 +0200
@@ -108,7 +108,7 @@
 #
 # how to build it info
 #
-USELIBS		= ../snmplib/libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
+USELIBS		= $(top_builddir)/snmplib/libnetsnmp.la
 AGENTLIB	= libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
 MIBLIB		= libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 
@@ -249,11 +249,11 @@
 
 
 libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION):    ${LLIBAGENTOBJS} $(USELIBS)
-	$(LIB_LD_CMD) $(AGENTLIB) ${LLIBAGENTOBJS} $(USELIBS) ${LAGENTLIBS} @LD_NO_UNDEFINED@ $(LDFLAGS) $(PERLLDOPTS_FOR_LIBS) $(LIB_LD_LIBS) @AGENTLIBS@
+	$(LIB_LD_CMD) $(AGENTLIB) ${LLIBAGENTOBJS} $(USELIBS) $(LDFLAGS) ${LAGENTLIBS} @LD_NO_UNDEFINED@ $(PERLLDOPTS_FOR_LIBS) $(LIB_LD_LIBS) @AGENTLIBS@
 	$(RANLIB) $(AGENTLIB)
 
 libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION): ${LMIBOBJS} $(AGENTLIB) $(USELIBS)
-	$(LIB_LD_CMD) $(MIBLIB) ${LMIBOBJS} $(AGENTLIB) $(USELIBS) @LD_NO_UNDEFINED@ $(LDFLAGS) ${LMIBLIBS} $(LIB_LD_LIBS) @AGENTLIBS@
+	$(LIB_LD_CMD) $(MIBLIB) ${LMIBOBJS} $(LDFLAGS) $(AGENTLIB) $(USELIBS) @LD_NO_UNDEFINED@ ${LMIBLIBS} $(LIB_LD_LIBS) @AGENTLIBS@
 	$(RANLIB) $(MIBLIB)
 
 agentlib: $(AGENTLIB)
--- net-snmp-5.4.1/Makefile.rules.orig	2007-07-07 01:39:10.000000000 +0200
+++ net-snmp-5.4.1/Makefile.rules	2007-09-08 16:43:38.055836050 +0200
@@ -8,7 +8,7 @@
 
 # the standard items to build: libraries, bins, and sbins
 STANDARDTARGETS     =$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS)
-STANDARDCLEANTARGETS=$(INSTALLLIBS) $(INSTALLPOSTLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS) $(INSTALLUCDLIBS)
+STANDARDCLEANTARGETS=$(INSTALLLIBS) $(INSTALLPOSTLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS)
 
 standardall: subdirs $(STANDARDTARGETS)
 
@@ -187,17 +187,11 @@
 		$(LIB_LDCONFIG_CMD) ; \
 	fi
 
-installucdlibs: $(INSTALLUCDLIBS)
-	@if test "$(INSTALLUCDLIBS)" != ""; then \
-		it="$(INSTALLUCDLIBS)" ; \
-		$(SHELL) $(top_srcdir)/mkinstalldirs $(INSTALL_PREFIX)$(libdir) ; \
-		$(INSTALL) $(INSTALLUCDLIBS) $(INSTALL_PREFIX)$(libdir) ; \
-		for i in $$it ; do \
-			echo "installing $$i in $(INSTALL_PREFIX)$(libdir)"; \
-			$(RANLIB) $(INSTALL_PREFIX)$(libdir)/$$i ; \
-		done ; \
-		$(LIB_LDCONFIG_CMD) ; \
-	fi
+installucdlibs:
+	$(SHELL) $(top_srcdir)/mkinstalldirs $(INSTALL_PREFIX)$(libdir) ; \
+	ln -sf libnetsnmp.so $(INSTALL_PREFIX)$(libdir)/libsnmp.so ; \
+	ln -sf libnetsnmp.la $(INSTALL_PREFIX)$(libdir)/libsnmp.la ; \
+	ln -sf libnetsnmp.a $(INSTALL_PREFIX)$(libdir)/libsnmp.a
 
 installsubdirlibs:
 	@if test "$(SUBDIRS)" != ""; then \
--- net-snmp-5.6/snmplib/Makefile.in.orig	2010-05-27 06:06:32.000000000 +0200
+++ net-snmp-5.6/snmplib/Makefile.in	2010-10-24 10:35:59.615743558 +0200
@@ -134,7 +134,6 @@
 
 # libraries
 INSTALLLIBS=libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
-INSTALLUCDLIBS=libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 #
 # Things to build
@@ -200,11 +199,7 @@
 
 # how to build the libraries.
 libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    $(TOBJS)
-	$(LIB_LD_CMD) $@ $(TOBJS) @LD_NO_UNDEFINED@ $(LDFLAGS) @LNETSNMPLIBS@
-	$(RANLIB) $@
-
-libsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    $(TOBJS)
-	$(LIB_LD_CMD) $@ $(TOBJS) @LD_NO_UNDEFINED@ $(LDFLAGS) @LNETSNMPLIBS@
+	$(LIB_LD_CMD) $@ $(TOBJS) @LD_NO_UNDEFINED@ $(LDFLAGS) @LNETSNMPLIBS@ @LIBS@
 	$(RANLIB) $@
 
 #
