--- net-snmp-5.8/Makefile.in.orig	2018-08-25 22:44:53.077693735 +0200
+++ net-snmp-5.8/Makefile.in	2018-08-25 22:50:13.384356740 +0200
@@ -234,7 +234,7 @@
         fi
 
 pythoninstall:
-	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir --root=$(DESTDIR) --prefix=$(prefix)) ; \
+	@(dir=`pwd`; cd python; $(PYMAKE) install --optimize=2 --basedir=$$dir --root=$(DESTDIR) --prefix=$(prefix)) ; \
         if test $$? != 0 ; then \
            exit 1 ; \
         fi
