diff -urNp -x '*.orig' net-snmp-5.9.org/man/mib2c.1.def net-snmp-5.9/man/mib2c.1.def
--- net-snmp-5.9.org/man/mib2c.1.def	2020-08-14 23:41:47.000000000 +0200
+++ net-snmp-5.9/man/mib2c.1.def	2021-02-28 14:28:48.770042705 +0100
@@ -71,7 +71,7 @@ Display a help message.
 .TP
 .BI -c " CONFIGFILE"
 Use CONFIGFILE when generating code.  These files will be searched for
-first in the current directory and then in the DATADIR directory
+first in the current directory and then in the DATADIR/snmp directory
 (which is where the default mib2c configuration files can be found).
 Running mib2c without the -c CONFIGFILE option will display
 a description of the valid values for CONFIGFILE, that is,
