--- net-snmp-5.4.1/configure.ac	2007-09-08 16:13:11.531748373 +0200
+++ net-snmp-5.4.1-ac/configure.ac	2007-11-14 20:58:16.361317122 +0200
@@ -1726,16 +1726,18 @@
 rebuilt_new_with_security=
 already_checked_krb5=no
 
+KRB5_LIBS=
+AC_SUBST(KRB5_LIBS)
 for sec in $new_with_security
 do
   if test "x$sec" = "xksm" -a "xyes" = "x$already_checked_krb5"; then
     continue
   fi
-  already_checked_krb5=yes
 
   if test x$sec != xksm; then
     rebuilt_new_with_security="$rebuilt_new_with_security $sec"
   else
+    already_checked_krb5=yes
     if test no = $trykrb5; then
       AC_MSG_NOTICE([configured not to use krb5. disabling ksm])
       continue
@@ -1745,7 +1747,7 @@
 
     if test no != $krb5_config; then
       CPPFLAGS="$CPPFLAGS `$krb5_config --cflags`"
-      LIBS="$LIBS `$krb5_config --libs`"
+      KRB5_LIBS="`$krb5_config --libs`"
     elif test $krb5path != no; then
       AC_ADD_SEARCH_PATH($krb5path)
     else
@@ -1782,29 +1784,35 @@
     if test $ac_cv_heimdal = yes; then
       AC_DEFINE(NETSNMP_USE_KERBEROS_HEIMDAL)
       if test no = $krb5_config; then
-	LIBS="$LIBS -lkrb5 -lasn1 -lcom_err -lroken"
+	KRB5_LIBS="-lkrb5 -lasn1 -lcom_err -lroken"
       fi
     elif test no = $krb5_config; then
-	LIBS="$LIBS -lkrb5 -lcom_err"
+	KRB5_LIBS="-lkrb5 -lcom_err"
     fi
 
+    old_LIBS="$LIBS"
+    LIBS="$LIBS $KRB5_LIBS"
     AC_MSG_CHECKING([to see which crypto API we need to use])
     AC_CHECK_FUNC(krb5_c_encrypt,[
-    AC_CACHE_VAL(ac_cv_NETSNMP_USE_KERBEROS_MIT,
-      AC_CHECK_LIB(k5crypto, krb5_init_context,
-	ac_cv_NETSNMP_USE_KERBEROS_MIT=yes, ac_cv_MIT_NEW_CRYPTo=no
-      )
-    )
+	AC_CHECK_LIB(k5crypto, krb5_init_context,
+		     ac_cv_MIT_NEW_CRYPTo=yes, ac_cv_MIT_NEW_CRYPTo=no
+	)
 
-    if test yes = $ac_cv_NETSNMP_USE_KERBEROS_MIT; then
-      AC_MSG_RESULT(new MIT crypto API)
+    if test yes = $ac_cv_MIT_NEW_CRYPTo; then
       AC_DEFINE(NETSNMP_USE_KERBEROS_MIT)
+      AC_MSG_RESULT(new MIT crypto API)
       if test no = $krb5_config; then
-	LIBS="$LIBS -lk5crypto"
+	KRB5_LIBS="$LIBS -lk5crypto"
+      else
+	CFLAGS="$CFLAGS `$krb5_config --cflags`"
+	KRB5_LIBS="$LIBS `$krb5_config --libs`"
       fi
     else
-      AC_MSG_RESULT(old MIT crypto API)])
+      AC_MSG_RESULT(old MIT crypto API)
     fi
+    ])
+    LNETSNMPLIBS="$LNETSNMPLIBS $KRB5_LIBS"
+    LIBS="$old_LIBS"
   fi
 done
 
