--- net-snmp-5.1/configure.in.orig	2003-11-18 20:33:31.000000000 +0100
+++ net-snmp-5.1/configure.in	2003-11-18 20:41:43.917724056 +0100
@@ -22,7 +22,7 @@
 #
 # save the configure arguments
 #
-AC_DEFINE_UNQUOTED(CONFIGURE_OPTIONS,"\"$ac_configure_args\"")
+AC_DEFINE_UNQUOTED(CONFIGURE_OPTIONS,"$ac_configure_args")
 CONFIGURE_OPTIONS="\"$ac_configure_args\""
 AC_SUBST(CONFIGURE_OPTIONS)
 
@@ -950,7 +950,6 @@
 
 if test "x$with_res_locks" = "xyes"; then
   AC_DEFINE(NS_REENTRANT)
-EOF
 
     case $target_os in
 
