--- net-snmp-5.2.1/net-snmp-config.in.orig	2005-01-05 16:12:52.000000000 +0100
+++ net-snmp-5.2.1/net-snmp-config.in	2005-02-26 14:58:33.779999232 +0100
@@ -46,7 +46,7 @@
 includedir=@includedir@
 libdir=@libdir@
 datarootdir=@datarootdir@
-NSC_LDFLAGS="@LDFLAGS@"
+NSC_LDFLAGS=""
 NSC_INCLUDEDIR=${includedir}
 NSC_LIBDIR=-L${libdir}
 NSC_LIBS="@LIBS@"
@@ -89,10 +89,10 @@
       ;;
     #################################################### compile
     --base-cflags)
-      echo @CFLAGS@ @CPPFLAGS@ -I${NSC_INCLUDEDIR}
+      echo @CPPFLAGS@ -I${NSC_INCLUDEDIR}
       ;;
     --cflags|--cf*)
-      echo @CFLAGS@ @DEVFLAGS@ @CPPFLAGS@ -I. -I${NSC_INCLUDEDIR}
+      echo @DEVFLAGS@ @CPPFLAGS@ -I. -I${NSC_INCLUDEDIR}
       ;;
     --srcdir)
       echo $NSC_SRCDIR
@@ -445,7 +445,7 @@
         echo "Ack.  Can't create $tmpfile."
 	exit 1
       fi
-      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS $ldflags"
+      cmd="@CC@ $cflags @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS $ldflags"
       echo "running: $cmd"
       `$cmd`
       if test "x$norm" != "x1" ; then
