--- net-snmp-5.5/agent/mibgroup/ucd-snmp/loadave.c~	2010-05-12 14:39:47.608334020 +0200
+++ net-snmp-5.5/agent/mibgroup/ucd-snmp/loadave.c	2010-05-12 14:40:38.068953645 +0200
@@ -166,7 +166,7 @@
          var_extensible_loadave, 1, {LOADAVE}},
         {LOADMAXVAL, ASN_OCTET_STR, NETSNMP_OLDAPI_RONLY,
          var_extensible_loadave, 1, {LOADMAXVAL}},
-        {LOADAVEINT, ASN_INTEGER, NETSNMP_OLDAPI_RONLY,
+        {LOADAVEINT, ASN_INTEGER, NETSNMP_OLDAPI_RWRITE,
          var_extensible_loadave, 1, {LOADAVEINT}},
 #ifdef NETSNMP_WITH_OPAQUE_SPECIAL_TYPES
         {LOADAVEFLOAT, ASN_OPAQUE_FLOAT, NETSNMP_OLDAPI_RONLY,
