diff -ruN net-snmp-5.3.1./agent/mibgroup/host/hr_swinst.c net-snmp-5.3.1/agent/mibgroup/host/hr_swinst.c
--- net-snmp-5.3.1./agent/mibgroup/host/hr_swinst.c	2006-03-09 18:31:53.000000000 +0100
+++ net-snmp-5.3.1/agent/mibgroup/host/hr_swinst.c	2006-08-07 23:19:42.356996008 +0200
@@ -205,7 +205,7 @@
         /*
          * XXX distinguish between rpm-2.5.x and rpm-2.9x 
          */
-#ifdef HAVE_RPMGETPATH
+#if 1
         rpmReadConfigFiles(NULL, NULL);
         swi->swi_dbpath = rpmGetPath("%{_dbpath}", NULL);
 #else
