--- net-snmp-5.2/include/net-snmp/types.h.orig	2004-09-14 04:29:15.000000000 +0200
+++ net-snmp-5.2/include/net-snmp/types.h	2005-01-17 21:54:03.359892800 +0100
@@ -77,34 +77,34 @@
      */
 #ifndef HAVE_INT32_T
 #if   SIZEOF_INT == 4
-typedef int int32_t
+typedef int int32_t;
 #elif SIZEOF_LONG == 4
-typedef long int32_t
+typedef long int32_t;
 #elif SIZEOF_SHORT == 4
-typedef short int32_t
+typedef short int32_t;
 #else
-typedef int int32_t
+typedef int int32_t;
 #define _INT32_IS_NOT_32BIT
 #endif
 #endif
 
 #ifndef HAVE_UINT32_T
 #ifdef HAVE_U_INT32_T
-typedef u_int32_t        uint32_t
+typedef u_int32_t        uint32_t;
 #else
-typedef unsigned int32_t uint32_t
+typedef unsigned int32_t uint32_t;
 #endif
 #endif
 
 #ifndef HAVE_INT64_T
 #if SIZEOF_LONG == 8
-typedef long int64_t
+typedef long int64_t;
 #elif SIZEOF_LONG_LONG == 8
-typedef long long int64_t
+typedef long long int64_t;
 #elif   SIZEOF_INT == 8
-typedef int int64_t
+typedef int int64_t;
 #elif SIZEOF_LONG >= 8
-typedef long int64_t
+typedef long int64_t;
 #define _INT64_IS_NOT_64BIT
 #else
 #define _NO_64BIT_TYPE 1
@@ -113,9 +113,9 @@
 
 #ifndef HAVE_UINT64_T
 #ifdef HAVE_U_INT64_T
-typedef u_int64_t        uint64_t
+typedef u_int64_t        uint64_t;
 #elif !defined(_NO_64BIT_TYPE)
-typedef unsigned int64_t uint64_t
+typedef unsigned int64_t uint64_t;
 #endif
 #endif
 
