diff -urN net-snmp-5.4.org/agent/Makefile.in net-snmp-5.4/agent/Makefile.in
--- net-snmp-5.4.org/agent/Makefile.in	2006-11-07 15:54:25.000000000 +0100
+++ net-snmp-5.4/agent/Makefile.in	2006-12-21 01:59:59.846638250 +0100
@@ -74,11 +74,11 @@
 #
 # how to build it info
 #
-USELIBS		= ../snmplib/libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
-HELPERLIB	= helpers/libnetsnmphelpers.$(LIB_EXTENSION)$(LIB_VERSION)
+USELIBS		= $(top_builddir)/snmplib/libnetsnmp.la
+HELPERLIB	= $(top_builddir)/agent/helpers/libnetsnmphelpers.la
 LOCAL_LIBS	= 
 PERLLDOPTS      = @PERLLDOPTS@
-LIBS		= ../snmplib/libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) @AGENTLIBS@ $(PERLLDOPTS)
+LIBS		= $(top_builddir)/snmplib/libnetsnmp.la @AGENTLIBS@ $(PERLLDOPTS)
 OUR_AGENT_LIBS	= $(INSTALLLIBS) $(HELPERLIB) @WRAPLIBS@ $(LIBS) @DLLIBS@
 CPPFLAGS	= $(TOP_INCLUDES) -I. $(AGENT_INCLUDES) $(MIBGROUP_INCLUDES) \
 		  $(SNMPLIB_INCLUDES) @CPPFLAGS@ 
@@ -137,11 +137,11 @@
 
 
 libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION):    ${LLIBAGENTOBJS}
-	$(LIB_LD_CMD) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS}  $(LIB_LD_LIBS)
+	$(LIB_LD_CMD) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS} @WRAPLIBS@ $(LIB_LD_LIBS) ${HELPERLIB} ${LIBS}
 	$(RANLIB) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
 
 libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION):    ${LMIBOBJS}
-	$(LIB_LD_CMD) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS}  $(LIB_LD_LIBS) @SENSORLIBS@
+	$(LIB_LD_CMD) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS} ${HELPERLIB} $(LIB_LD_LIBS) @SENSORLIBS@ -ldl
 	$(RANLIB) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 
 agentlib: libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
diff -urN net-snmp-5.4.org/configure.in net-snmp-5.4/configure.in
--- net-snmp-5.4.org/configure.in	2006-11-24 18:53:18.000000000 +0100
+++ net-snmp-5.4/configure.in	2006-12-21 01:17:16.166418250 +0100
@@ -1699,6 +1699,8 @@
 rebuilt_new_with_security=
 already_checked_krb5=no
 
+KRB5_LIBS=
+AC_SUBST(KRB5_LIBS)
 for sec in $new_with_security
 do
   if test "x$sec" = "xksm" -a "xyes" = "x$already_checked_krb5"; then
@@ -1718,7 +1720,7 @@
 
     if test no != $krb5_config; then
       CPPFLAGS="$CPPFLAGS `$krb5_config --cflags`"
-      LIBS="$LIBS `$krb5_config --libs`"
+      KRB5_LIBS="`$krb5_config --libs`"
     elif test $krb5path != no; then
       AC_ADD_SEARCH_PATH($krb5path)
     else
@@ -1755,10 +1757,10 @@
     if test $ac_cv_heimdal = yes; then
       AC_DEFINE(NETSNMP_USE_KERBEROS_HEIMDAL)
       if test no = $krb5_config; then
-	LIBS="$LIBS -lkrb5 -lasn1 -lcom_err -lroken"
+	KRB5_LIBS="-lkrb5 -lasn1 -lcom_err -lroken"
       fi
     elif test no = $krb5_config; then
-	LIBS="$LIBS -lkrb5 -lcom_err"
+	KRB5_LIBS="-lkrb5 -lcom_err"
     fi
 
     AC_MSG_CHECKING([to see which crypto API we need to use])
@@ -1773,7 +1775,7 @@
       AC_MSG_RESULT(new MIT crypto API)
       AC_DEFINE(NETSNMP_USE_KERBEROS_MIT)
       if test no = $krb5_config; then
-	LIBS="$LIBS -lk5crypto"
+	KRB5_LIBS="-lk5crypto"
       fi
     else
       AC_MSG_RESULT(old MIT crypto API)])
diff -urN net-snmp-5.4.org/Makefile.rules net-snmp-5.4/Makefile.rules
--- net-snmp-5.4.org/Makefile.rules	2005-10-10 20:47:48.000000000 +0200
+++ net-snmp-5.4/Makefile.rules	2006-12-21 01:17:16.166418250 +0100
@@ -8,7 +8,7 @@
 
 # the standard items to build: libraries, bins, and sbins
 STANDARDTARGETS     =$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS)
-STANDARDCLEANTARGETS=$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS) $(INSTALLUCDLIBS)
+STANDARDCLEANTARGETS=$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS)
 
 standardall: subdirs $(STANDARDTARGETS)
 
@@ -175,17 +175,11 @@
 		$(LIB_LDCONFIG_CMD) ; \
 	fi
 
-installucdlibs: $(INSTALLUCDLIBS)
-	@if test "$(INSTALLUCDLIBS)" != ""; then \
-		it="$(INSTALLUCDLIBS)" ; \
-		$(SHELL) $(top_srcdir)/mkinstalldirs $(INSTALL_PREFIX)$(libdir) ; \
-		$(INSTALL) $(INSTALLUCDLIBS) $(INSTALL_PREFIX)$(libdir) ; \
-		for i in $$it ; do \
-			echo "installing $$i in $(INSTALL_PREFIX)$(libdir)"; \
-			$(RANLIB) $(INSTALL_PREFIX)$(libdir)/$$i ; \
-		done ; \
-		$(LIB_LDCONFIG_CMD) ; \
-	fi
+installucdlibs:
+	$(SHELL) $(top_srcdir)/mkinstalldirs $(INSTALL_PREFIX)$(libdir) ; \
+	ln -sf libnetsnmp.so $(INSTALL_PREFIX)$(libdir)/libsnmp.so ; \
+	ln -sf libnetsnmp.la $(INSTALL_PREFIX)$(libdir)/libsnmp.la ; \
+	ln -sf libnetsnmp.a $(INSTALL_PREFIX)$(libdir)/libsnmp.a
 
 installsubdirlibs:
 	@if test "$(SUBDIRS)" != ""; then \
diff -urN net-snmp-5.4.org/snmplib/Makefile.in net-snmp-5.4/snmplib/Makefile.in
--- net-snmp-5.4.org/snmplib/Makefile.in	2006-09-19 16:45:29.000000000 +0200
+++ net-snmp-5.4/snmplib/Makefile.in	2006-12-21 01:17:16.166418250 +0100
@@ -128,7 +128,6 @@
 
 # libraries
 INSTALLLIBS=libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
-INSTALLUCDLIBS=libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 #
 # Things to build
@@ -188,13 +187,9 @@
 
 # how to build the libraries.
 libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} ${LDFLAGS} @LIBCRYPTO@
+	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} ${LDFLAGS} @KRB5_LIBS@ @LIBCRYPTO@
 	$(RANLIB) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
-libsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} ${LDFLAGS} @LIBCRYPTO@
-	$(RANLIB) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
-
 #
 # internal test objects
 #
