diff -durN -x '*~' net-snmp-5.1.orig/agent/mibgroup/tunnel/tunnel.c net-snmp-5.1/agent/mibgroup/tunnel/tunnel.c
--- net-snmp-5.1.orig/agent/mibgroup/tunnel/tunnel.c	2002-10-02 19:59:07.000000000 +0200
+++ net-snmp-5.1/agent/mibgroup/tunnel/tunnel.c	2004-12-21 16:41:08.301032640 +0100
@@ -53,11 +53,11 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
-#include <linux/if.h>
-#include <linux/ip.h>
+#include <net/if.h>
+#include <netinet/ip.h>
 #include <linux/sockios.h>
 #include <linux/if_tunnel.h>
-#include <linux/if_arp.h>
+#include <net/if_arp.h>
 
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
