diff -ur net-snmp-5.3.org/agent/Makefile.in net-snmp-5.3/agent/Makefile.in
--- net-snmp-5.3.org/agent/Makefile.in	2005-12-04 19:24:25.000000000 +0100
+++ net-snmp-5.3/agent/Makefile.in	2006-01-09 00:58:58.598965000 +0100
@@ -133,7 +133,7 @@
 
 
 libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION):    ${LLIBAGENTOBJS}
-	$(LIB_LD_CMD) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS}  $(LIB_LD_LIBS)
+	$(LIB_LD_CMD) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS} @WRAPLIBS@ $(LIB_LD_LIBS)
 	$(RANLIB) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
 
 libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION):    ${LMIBOBJS}
diff -ur net-snmp-5.3.org/Makefile.rules net-snmp-5.3/Makefile.rules
--- net-snmp-5.3.org/Makefile.rules	2005-10-10 20:47:48.000000000 +0200
+++ net-snmp-5.3/Makefile.rules	2006-01-09 00:58:13.044118000 +0100
@@ -8,7 +8,7 @@
 
 # the standard items to build: libraries, bins, and sbins
 STANDARDTARGETS     =$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS)
-STANDARDCLEANTARGETS=$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS) $(INSTALLUCDLIBS)
+STANDARDCLEANTARGETS=$(INSTALLLIBS) $(INSTALLBINPROGS) $(INSTALLSBINPROGS)
 
 standardall: subdirs $(STANDARDTARGETS)
 
@@ -175,17 +175,11 @@
 		$(LIB_LDCONFIG_CMD) ; \
 	fi
 
-installucdlibs: $(INSTALLUCDLIBS)
-	@if test "$(INSTALLUCDLIBS)" != ""; then \
-		it="$(INSTALLUCDLIBS)" ; \
-		$(SHELL) $(top_srcdir)/mkinstalldirs $(INSTALL_PREFIX)$(libdir) ; \
-		$(INSTALL) $(INSTALLUCDLIBS) $(INSTALL_PREFIX)$(libdir) ; \
-		for i in $$it ; do \
-			echo "installing $$i in $(INSTALL_PREFIX)$(libdir)"; \
-			$(RANLIB) $(INSTALL_PREFIX)$(libdir)/$$i ; \
-		done ; \
-		$(LIB_LDCONFIG_CMD) ; \
-	fi
+installucdlibs:
+	$(SHELL) $(top_srcdir)/mkinstalldirs $(INSTALL_PREFIX)$(libdir) ; \
+	ln -sf libnetsnmp.so $(INSTALL_PREFIX)$(libdir)/libsnmp.so ; \
+	ln -sf libnetsnmp.la $(INSTALL_PREFIX)$(libdir)/libsnmp.la ; \
+	ln -sf libnetsnmp.a $(INSTALL_PREFIX)$(libdir)/libsnmp.a
 
 installsubdirlibs:
 	@if test "$(SUBDIRS)" != ""; then \
diff -ur net-snmp-5.3.org/snmplib/Makefile.in net-snmp-5.3/snmplib/Makefile.in
--- net-snmp-5.3.org/snmplib/Makefile.in	2005-10-30 20:01:14.000000000 +0100
+++ net-snmp-5.3/snmplib/Makefile.in	2006-01-09 00:58:13.044118000 +0100
@@ -127,7 +127,6 @@
 
 # libraries
 INSTALLLIBS=libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
-INSTALLUCDLIBS=libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 #
 # Things to build
@@ -187,13 +186,9 @@
 
 # how to build the libraries.
 libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS}
+	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} -lcrypto
 	$(RANLIB) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
-libsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS}
-	$(RANLIB) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
-
 #
 # internal test objects
 #
