diff -ur net-snmp-5.3.org/configure.in net-snmp-5.3/configure.in
--- net-snmp-5.3.org/configure.in	2005-12-28 18:20:56.000000000 +0100
+++ net-snmp-5.3/configure.in	2006-01-09 01:07:39.031490000 +0100
@@ -1704,11 +1704,11 @@
   if test "x$sec" = "xksm" -a "xyes" = "x$already_checked_krb5"; then
     continue
   fi
-  already_checked_krb5=yes
 
   if test x$sec != xksm; then
     rebuilt_new_with_security="$rebuilt_new_with_security $sec"
   else
+    already_checked_krb5=yes
     if test no = $trykrb5; then
       AC_MSG_NOTICE([configured not to use krb5. disabling ksm])
       continue
@@ -1763,15 +1763,13 @@
 
     AC_MSG_CHECKING([to see which crypto API we need to use])
     AC_CHECK_FUNC(krb5_c_encrypt,[
-    AC_CACHE_VAL(ac_cv_NETSNMP_USE_KERBEROS_MIT,
-      AC_CHECK_LIB(k5crypto, krb5_init_context,
-	ac_cv_NETSNMP_USE_KERBEROS_MIT=yes, ac_cv_MIT_NEW_CRYPTo=no
-      )
-    )
+	AC_CHECK_LIB(k5crypto, krb5_init_context,
+		     ac_cv_MIT_NEW_CRYPTo=yes, ac_cv_MIT_NEW_CRYPTo=no
+	)
 
-    if test yes = $ac_cv_NETSNMP_USE_KERBEROS_MIT; then
+    AC_DEFINE(NETSNMP_USE_KERBEROS_MIT)
+    if test yes = $ac_cv_MIT_NEW_CRYPTo; then
       AC_MSG_RESULT(new MIT crypto API)
-      AC_DEFINE(NETSNMP_USE_KERBEROS_MIT)
       if test no = $krb5_config; then
 	LIBS="$LIBS -lk5crypto"
       fi
@@ -2661,23 +2661,6 @@
     LIBS="${LIBS} -ladm"
     AGENTLIBS="${AGENTLIBS} -ladm"))
 
-case $target_os in
-    aix*) # AIX perfstat library, needed for CPU/memory statistics
-        AC_CHECK_HEADERS(libperfstat.h,
-            AC_CHECK_LIB(perfstat, perfstat_cpu_total, AC_DEFINE(HAVE_PERFSTAT)
-            AGENTLIBS="${AGENTLIBS} -lperfstat"), AC_MSG_ERROR([
-
-*** To monitor CPU/memory values in AIX you need to install
-*** libperfstat which can be found in bos.perf
-]))
-        ;;
-    hpux*) # HP-UX agent needs open_mib
-        AC_CHECK_LIB(nm, open_mib, AC_DEFINE(HAVE_LIBNM)
-            AGENTLIBS="${AGENTLIBS} -lnm")
-        ;;
-    *)
-        ;;
-esac
 # WIN32
 # (mingw32 must link winsock explicitly)
 AC_CHECK_HEADERS(winsock.h,[
@@ -2880,7 +2863,7 @@
 
 AC_MSG_RESULT([$ac_cv_type_ssize_t])
 if test $ac_cv_type_ssize_t = yes; then
-  AC_DEFINE(HAVE_SSIZE_T)
+  AC_DEFINE(HAVE_SSIZE_T,,[have ssize_t])
 fi
 
 # Check ps args
--- net-snmp-5.4/agent/mibgroup/Makefile.in.orig	2006-09-13 19:00:27.000000000 +0200
+++ net-snmp-5.4/agent/mibgroup/Makefile.in	2006-12-02 11:51:38.369040000 +0100
@@ -22,9 +22,13 @@
 .c.o:
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
 
-OBJS  = @mib_module_list_o@
-LOBJS = @mib_module_list_lo@
-SRCS  = @mib_module_list_c@
+@mib_module_list_o@
+@mib_module_list_lo@
+@mib_module_list_c@
+
+OBJS  = $(mib_module_list_o)
+LOBJS = $(mib_module_list_lo)
+SRCS  = $(mib_module_list_c)
 
 all: standardall $(LOBJS)
 

