diff -urNp -x '*.orig' net-snmp-5.9.org/agent/mibgroup/host/data_access/swinst.h net-snmp-5.9/agent/mibgroup/host/data_access/swinst.h
--- net-snmp-5.9.org/agent/mibgroup/host/data_access/swinst.h	2020-08-14 23:41:47.000000000 +0200
+++ net-snmp-5.9/agent/mibgroup/host/data_access/swinst.h	2021-02-28 14:29:14.950860864 +0100
@@ -11,6 +11,10 @@
 #ifndef NETSNMP_ACCESS_SWINST_CONFIG_H
 #define NETSNMP_ACCESS_SWINST_CONFIG_H
 
+// quick fix: this file is consulted before RPM_HRMIB_PATH check is made in configure, hardcode a define for now
+#define RPM_HRMIB_PATH "/var/cache/hrmib"
+
+
 /*
  * all platforms use this generic code
  */
@@ -25,7 +29,7 @@ config_exclude(host/hr_swinst)
     config_require(host/data_access/swinst_apt)
 #elif defined( HAVE_LIBRPM ) && defined( linux )
     config_require(host/data_access/swinst_rpm)
-#elif defined( HAVE_PKGLOCS_H ) || defined( hpux9 ) || defined( hpux10 ) || defined( hpux11 ) || defined( freebsd2 ) || defined( linux ) || defined( openbsd )
+#elif defined( HAVE_PKGLOCS_H ) || defined( RPM_HRMIB_PATH ) || defined( hpux9 ) || defined( hpux10 ) || defined( hpux11 ) || defined( freebsd2 ) || defined( linux ) || defined( openbsd )
     config_require(host/data_access/swinst_pkginfo)
 #else
     config_warning(This platform does not yet support hrSWInstalledTable rewrites)
diff -urNp -x '*.orig' net-snmp-5.9.org/agent/mibgroup/host/data_access/swinst_pkginfo.c net-snmp-5.9/agent/mibgroup/host/data_access/swinst_pkginfo.c
--- net-snmp-5.9.org/agent/mibgroup/host/data_access/swinst_pkginfo.c	2020-08-14 23:41:47.000000000 +0200
+++ net-snmp-5.9/agent/mibgroup/host/data_access/swinst_pkginfo.c	2021-02-28 14:29:14.950860864 +0100
@@ -61,6 +61,8 @@ netsnmp_swinst_arch_init(void)
 {
 #if defined(PKGLOC)
     snprintf( pkg_directory, SNMP_MAXPATH, "%s", PKGLOC );
+#elif defined(RPM_HRMIB_PATH)
+    snprintf( pkg_directory, SNMP_MAXPATH, "%s", RPM_HRMIB_PATH);
 #elif defined(hpux9)
     snprintf( pkg_directory, SNMP_MAXPATH, "/system" );
 #elif defined(hpux10) ||  defined(hpux11) 
diff -urNp -x '*.orig' net-snmp-5.9.org/agent/mibgroup/host/hr_swinst.c net-snmp-5.9/agent/mibgroup/host/hr_swinst.c
--- net-snmp-5.9.org/agent/mibgroup/host/hr_swinst.c	2021-02-28 14:29:14.640851176 +0100
+++ net-snmp-5.9/agent/mibgroup/host/hr_swinst.c	2021-02-28 14:29:14.950860864 +0100
@@ -196,6 +196,10 @@ oid             hrswinst_variables_oid[]
 #define	_PATH_HRSW_directory	"/var/cache/hrmib"
 #endif
 
+#if defined(RPM_HRMIB_PATH)
+#define	_PATH_HRSW_directory	RPM_HRMIB_PATH
+#endif
+
 void
 init_hr_swinst(void)
 {
diff -urNp -x '*.orig' net-snmp-5.9.org/configure.d/config_os_functions net-snmp-5.9/configure.d/config_os_functions
--- net-snmp-5.9.org/configure.d/config_os_functions	2020-08-14 23:41:47.000000000 +0200
+++ net-snmp-5.9/configure.d/config_os_functions	2021-02-28 14:29:14.950860864 +0100
@@ -188,15 +188,6 @@ AC_CHECK_FUNCS([if_freenameindex
 #           (Agent)
 ##
 
-#       rpmGetPath check needs clean LIBS environment   (Linux)
-#
-if test "x$with_rpm" != "xno"; then
-    OLDLIBS=$LIBS
-    LIBS=$LMIBLIBS
-    AC_CHECK_FUNCS(rpmGetPath)
-    LIBS=$OLDLIBS
-fi
-
 #       statfs( char *, struct fs_data* )               (Ultrix)
 #
 AC_MSG_CHECKING([[for two-argument statfs with struct fs_data (Ultrix)]])
diff -urNp -x '*.orig' net-snmp-5.9.org/configure.d/config_os_libs1 net-snmp-5.9/configure.d/config_os_libs1
--- net-snmp-5.9.org/configure.d/config_os_libs1	2021-02-28 14:29:14.630850863 +0100
+++ net-snmp-5.9/configure.d/config_os_libs1	2021-02-28 14:29:14.950860864 +0100
@@ -11,18 +11,33 @@
 #   RPM checks
 #
 ##
-AC_CHECK_HEADERS(rpm/rpmlib.h rpm/header.h)
 
-AC_CHECK_HEADER(rpm/rpmlib.h,rpmlib_h=yes,rpmlib_h=no)
-AC_CHECK_HEADER(rpm/header.h,rpmheader_h=yes,rpmheader_h=no)
-
-if test "x$rpmheader_h" = "xno" -o "x$rpmlib_h" = "xno" ; then
-  if test "x$with_rpm" = "xyes" ; then
-    AC_MSG_ERROR([Could not find either the RPM header files needed and was specifically asked to use RPM support])
-  else
-    with_rpm=no
-  fi
+# check if we can use rpm hrmib export
+if test "$with_rpm" != "no" ; then
+       AC_PATH_PROG([RPM], rpm)
+       if test -n "$RPM"; then
+               changequote(, )
+               RPM_HRMIB_PATH=$($RPM -E %{_hrmib_path} | sed -e 's,/[^/]\+$,,')
+               changequote([, ])
+               if test -n "$RPM_HRMIB_PATH"; then
+                       AC_DEFINE_UNQUOTED([RPM_HRMIB_PATH], ["$RPM_HRMIB_PATH"], [Directory where RPM exports packages hrmib information])
+               fi
+       fi
 fi
+# otherwise try binary linkin
+if test -z "$RPM_HRMIB_PATH"; then
+       AC_CHECK_HEADERS(rpm/rpmlib.h rpm/header.h)
+
+       AC_CHECK_HEADER(rpm/rpmlib.h,rpmlib_h=yes,rpmlib_h=no)
+       AC_CHECK_HEADER(rpm/header.h,rpmheader_h=yes,rpmheader_h=no)
+
+       if test "x$rpmheader_h" = "xno" -o "x$rpmlib_h" = "xno" ; then
+         if test "x$with_rpm" = "xyes" ; then
+               AC_MSG_ERROR([Could not find either the RPM header files needed and was specifically asked to use RPM support])
+         else
+               with_rpm=no
+         fi
+       fi
 
 #
 # rpm libraries only needed for the host resources mib software
@@ -46,6 +61,12 @@ if test "x$with_rpm" != "xno"; then
   NETSNMP_SEARCH_LIBS(headerGet,, [
     AC_DEFINE(HAVE_HEADERGET, 1,
       [Define to 1 if you have the headerGet function.])],, $LMIBLIBS)
+
+  OLDLIBS=$LIBS
+  LIBS=$LMIBLIBS
+  AC_CHECK_FUNCS([rpmGetPath])
+  LIBS=$OLDLIBS
+fi
 fi
 
 #
