--- net-snmp-5.1/net-snmp-config.in.orig	2003-10-14 18:15:47.000000000 +0200
+++ net-snmp-5.1/net-snmp-config.in	2003-11-18 20:29:20.152793512 +0100
@@ -29,7 +29,7 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-NSC_LDFLAGS="@LDFLAGS@"
+NSC_LDFLAGS=""
 NSC_LIBDIR=-L@libdir@
 NSC_LIBS="@LIBS@"
 NSC_AGENTLIBS="@AGENTLIBS@"
@@ -66,10 +66,10 @@
       ;;
     #################################################### compile
     --base-cflags)
-      echo @CFLAGS@ @CPPFLAGS@ -I@includedir@
+      echo @CPPFLAGS@ -I@includedir@
       ;;
     --cflags)
-      echo @CFLAGS@ @DEVFLAGS@ @CPPFLAGS@ -I. -I@includedir@
+      echo @DEVFLAGS@ @CPPFLAGS@ -I. -I@includedir@
       ;;
     --srcdir)
       echo $NSC_SRCDIR
@@ -409,7 +409,7 @@
         echo "Ack.  Can't create $tmpfile."
 	exit 1
       fi
-      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I@includedir@ -o $outname $tmpfile $cfiles $NSC_LDFLAGS -L$NSC_EXEC_PREFIX/lib -lnetsnmpagent -lnetsnmphelpers -lnetsnmpmibs -lnetsnmp $NSC_AGENTLIBS $ldflags"
+      cmd="@CC@ $cflags @DEVFLAGS@ -I. -I@includedir@ -o $outname $tmpfile $cfiles $NSC_LDFLAGS -L$NSC_EXEC_PREFIX/lib -lnetsnmpagent -lnetsnmphelpers -lnetsnmpmibs -lnetsnmp $NSC_AGENTLIBS $ldflags"
       echo "running: $cmd"
       `$cmd`
       if test "x$norm" != "x1" ; then
