diff -ur net-snmp-5.3.org/configure.in net-snmp-5.3/configure.in
--- net-snmp-5.3.org/configure.in	2005-12-28 18:20:56.000000000 +0100
+++ net-snmp-5.3/configure.in	2006-01-09 00:57:24.173063750 +0100
@@ -2396,20 +2396,6 @@
   ])
 
 
-  # two variants of db1 need checking.
-  AC_CHECK_LIB(db1, dbopen, [_rpmlibs="-ldb1 $_rpmlibs"],
-     AC_CHECK_LIB(db, dbopen, [_rpmlibs="-ldb $_rpmlibs"]))
-
-  # two variants of db3 need checking.
-  AC_CHECK_LIB(db-3.1, db_create, [_rpmlibs="-ldb-3.1 $_rpmlibs"],
-    AC_CHECK_LIB(db-3.0, db_create, [_rpmlibs="-ldb-3.0 $_rpmlibs"]))
-  
-  # rpm-3.0.5 and later needs popt.
-  AC_CHECK_LIB(popt, poptParseArgvString, [_rpmlibs="-lpopt $_rpmlibs"])
-  
-  # rpm-4.0.x needs rpmio.
-  AC_CHECK_LIB(rpmio, Fopen, [_rpmlibs="-lrpmio $_rpmlibs"],,$_rpmlibs)
-  
   # now check for rpm using the appropriate libraries.
   AC_CHECK_LIB(rpm, rpmGetFilesystemList,[
     AC_DEFINE(HAVE_LIBRPM)
