--- net-snmp-5.9/configure.d/config_os_libs2~	2020-08-14 23:41:47.000000000 +0200
+++ net-snmp-5.9/configure.d/config_os_libs2	2021-10-05 22:08:41.470701151 +0200
@@ -338,7 +338,7 @@
             LIBS="$netsnmp_save_LIBS"
         fi
         netsnmp_save_LIBS="$LIBS"
-        LIBS="-lssl"
+        LIBS="-lssl -lcrypto"
         AC_CHECK_FUNCS([TLS_method TLSv1_method DTLS_method DTLSv1_method]dnl
                        [SSL_library_init SSL_load_error_strings]dnl
 		       [ERR_get_error_all])
--- net-snmp-5.9/snmplib/snmp_openssl.c~	2020-08-14 23:41:47.000000000 +0200
+++ net-snmp-5.9/snmplib/snmp_openssl.c	2021-10-05 22:13:02.946834772 +0200
@@ -901,8 +901,8 @@
     unsigned long err;
     for (err = ERR_get_error(); err; err = ERR_get_error()) {
         snmp_log(LOG_ERR,"%s: %ld\n", prefix ? prefix: "openssl error", err);
-        snmp_log(LOG_ERR, "library=%d, function=%d, reason=%d\n",
-                 ERR_GET_LIB(err), ERR_GET_FUNC(err), ERR_GET_REASON(err));
+        snmp_log(LOG_ERR, "library=%d, reason=%d\n",
+                 ERR_GET_LIB(err), ERR_GET_REASON(err));
     }
 }
 #endif /* NETSNMP_FEATURE_REMOVE_OPENSSL_ERR_LOG */
